calendar-bundle/src/Resources/contao/classes/Calendar.php

```diff
@@ -227,6 +228,9 @@ protected function generateFiles($arrFeed)
 						break 3;
 					}
 
+					// Override the global page object (#2946)
+					$objPage = CalendarModel::findByPk($event['pid'])->getRelated('jumpTo');
```

Comment:
IMO using `$GLOBALS` instead of `global $objPage` would be better here for understanding the code. But not sure if there is a code style to follow here @leofeyer ?
```suggestion
					$GLOBALS['objPage'] = CalendarModel::findByPk($event['pid'])->getRelated('jumpTo');
```